src/Plugin/facets/query_type/LocalGovDirectoriesQueryType.php

```diff
@@ -87,4 +125,78 @@ public function build() {
     return $this->facet;
   }
 
+  /**
+   * Get facets from a facet group except it's own with a corresponding tag
+   * 
+   * Re-run the search api query, this time removing all the other facet groups
+   * except the one specified by the filter tag. The purpose of this is to get
+   * the other facets that would be limited by setting of the passed in groups
+   * facets. It allows us to see the other possible facets that could be 
+   * selected as part of an 'OR' group 'AND' the facets selected in the passed
+   * in group.
+   *
+   * @param String $filter_tag
+   *   The search api tag of the facet group.
+   * @return Array
+   *   Search api query facet results.
+   */
+  protected function getResultingFacetsFromFacetGroupExceptOwn(String $filter_tag):Array {
+
+    // Set up a special filter query which needs to clone the original.
+    $filter_query = clone $this->query->getOriginalQuery();
+    $filter_query->preExecute();
+
+    // Find conditions
+    $conditions = &$filter_query->getConditionGroup()->getConditions();
+
+    // Store removed conditions so we can reset them.
+    $removed_conditions = [];
+
+    // Loop through each conditions, removing ones that are not this filter tag.
+    foreach ($conditions as $tag => $condition) {
+      if ($condition instanceof ConditionGroupInterface) {
+        $tags = $condition->getTags();
+
+        // @todo Check that we are only removing facet conditions.
+        if (!in_array($filter_tag, $tags)) {
+
+          // Store the removed conditions and remove it.
+          $removed_conditions[$tag] = $conditions[$tag];
+          unset($conditions[$tag]);
+        }
+      }
+    }
+
+    // Execute the filter query and get the facets returned.
+    $filter_query->execute();
+    $facets = $filter_query->getResults()->getExtraData('search_api_facets');
+    $filter_query->postExecute();
+
+    // Beacuse for reasons unknown, removing the conditions removes it from all
+    // the following queries, even though we are fetching the original query.
+    // We can get around that by readding the conditions back in now that we 
+    // have the facets we want.
+    $conditions = array_merge($conditions, $removed_conditions);
+
+    // Since we will get every facet from the passed in facet group,
+    // we need to filter those out so checks with other facet groups will
+    // show us only the ones that are reachable from this facet group.
+    $facet_type_id = substr($filter_tag, 39);
+    $group_facet_ids = \Drupal::entityTypeManager()
+      ->getStorage('localgov_directories_facets')
+      ->getQuery()
+      ->condition('bundle', $facet_type_id)
+      ->execute();
+    $found_facets = $facets['localgov_directory_facets_filter'] ?? [];
+    $found_facets = array_filter($found_facets, function($item) use($group_facet_ids) {
+      if (!in_array(intval(trim($item['filter'], '"')), $group_facet_ids)) {
```

Comment:
If we do decide to reverse the facet filtering, then only filter this groups facets instead of here where we filter every other group.